In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import datetime 
import os

In [2]:
df_raw = pd.read_csv('Global YouTube Statistics.csv', encoding='latin-1')
df_raw

rank                    Youtuber  subscribers   video views  \
0       1                    T-Series    245000000  2.280000e+11   
1       2              YouTube Movies    170000000  0.000000e+00   
2       3                     MrBeast    166000000  2.836884e+10   
3       4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4       5                   SET India    159000000  1.480000e+11   
..    ...                         ...          ...           ...   
990   991               Natan por Aï¿     12300000  9.029610e+09   
991   992    Free Fire India Official     12300000  1.674410e+09   
992   993                       Panda     12300000  2.214684e+09   
993   994                 RobTopGames     12300000  3.741235e+08   
994   995                Make Joke Of     12300000  2.129774e+09   

             category                       Title  uploads         Country  \
0               Music                    T-Series    20082           India   
1    Film & Animation               youtubemovies        1   United States   
2       Entertainment                     MrBeast      741   United States   
3           Education  Cocomelon - Nursery Rhymes      966   United States   
4               Shows                   SET India   116536           India   
..                ...                         ...      ...             ...   
990            Sports               Natan por Aï¿     1200          Brazil   
991    People & Blogs    Free Fire India Official     1500           India   
992               NaN                 HybridPanda     2452  United Kingdom   
993            Gaming                 RobTopGames       39          Sweden   
994            Comedy                Make Joke Of       62           India   

    Abbreviation   channel_type  ...  subscribers_for_last_30_days  \
0             IN          Music  ...                     2000000.0   
1             US          Games  ...                           NaN   
2             US  Entertainment  ...                     8000000.0   
3             US      Education  ...                     1000000.0   
4             IN  Entertainment  ...                     1000000.0   
..           ...            ...  ...                           ...   
990           BR  Entertainment  ...                      700000.0   
991           IN          Games  ...                      300000.0   
992           GB          Games  ...                        1000.0   
993           SE          Games  ...                      100000.0   
994           IN         Comedy  ...                      100000.0   

     created_year  created_month  created_date  \
0          2006.0            Mar          13.0   
1          2006.0            Mar           5.0   
2          2012.0            Feb          20.0   
3          2006.0            Sep           1.0   
4          2006.0            Sep          20.0   
..            ...            ...           ...   
990        2017.0            Feb          12.0   
991        2018.0            Sep          14.0   
992        2006.0            Sep          11.0   
993        2012.0            May           9.0   
994        2017.0            Aug           1.0   

     Gross tertiary education enrollment (%)    Population  Unemployment rate  \
0                                       28.1  1.366418e+09               5.36   
1                                       88.2  3.282395e+08              14.70   
2                                       88.2  3.282395e+08              14.70   
3                                       88.2  3.282395e+08              14.70   
4                                       28.1  1.366418e+09               5.36   
..                                       ...           ...                ...   
990                                     51.3  2.125594e+08              12.08   
991                                     28.1  1.366418e+09               5.36   
992                                     60.0  6.683440e+07               3.85   

In [3]:
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 995 entries, 0 to 994
Data columns (total 28 columns):
 #   Column                                   Non-Null Count  Dtype  
---  ------                                   --------------  -----  
 0   rank                                     995 non-null    int64  
 1   Youtuber                                 995 non-null    object 
 2   subscribers                              995 non-null    int64  
 3   video views                              995 non-null    float64
 4   category                                 949 non-null    object 
 5   Title                                    995 non-null    object 
 6   uploads                                  995 non-null    int64  
 7   Country                                  873 non-null    object 
 8   Abbreviation                             873 non-null    object 
 9   channel_type                             965 non-null    object 
 10  video_views_rank                         994 non-n

# Column Standardization

To improve the structure of the dataset, we made the following modifications:

### Rename columns with spaces:

Rename the following columns to remove spaces and improve consistency:

- "Video Views" was changed to "video_views."
- "Gross tertiary education enrollment (%)" was changed to "percentage_tertiary_education."
- "Unemployment rate" was changed to "unemployment_rate."

In [4]:
df_raw.rename(columns = {'video views':'video_views','Gross tertiary education enrollment (%)': 'percentage_tertiary_education', 'Unemployment rate':'unemployment_rate'}, inplace= True)

### Rename columns with Uppercase Letters

Renamed the following columns so that all initial letters are in lowercase, following the recommended naming style:

- “Youtuber” was changed to “youtuber.”
- "Title" was changed to "title."
- "Country" was changed to "country."
- "Abbreviation" was changed to "abbreviation."
- "Population" was changed to "population."
- "Urban_population" was changed to "urban_population."
- "Latitude" was changed to "latitude."
- "Longitude" was changed to "longitude."

In [5]:
df_raw.rename(columns= {'Youtuber': 'youtuber', 'Title':'title', 'Country':'country', 'Abbreviation': 'abbreviation', 'Population': 'population', 'Urban_population':'urban_population', 'Latitude':'latitude', 'Longitude':'longitude'}, inplace = True)

# Duplicate information

 **'youtuber' and 'title' columns**
- The columns contained duplicate information for YouTube channel names.
1. Removed the 'title' column because the youtuber column had a more suitable and consistent format compared to 'title'.
2. Rename the 'youtuber' colum to 'channel_name'

In [6]:
df_raw.drop('title', axis= 1, inplace = True)

In [7]:
df_raw.rename(columns= {'youtuber': 'channel_name'}, inplace= True)

 **'category' and 'channel_type' columns**
 - The columns contained similar information but slightly differently named information.
1. Filling Null Values in 'category' with  values from the 'channel_type' column
2. Created a mapping dictionary that related similar categories with different names
3. Replacing the values in the 'category' column with values from the 'equivalents_category' Dictionary
 

In [8]:
df_raw['category'].fillna(df_raw['channel_type'], inplace = True)

In [9]:
equivalents_category={
   'Games': 'Gaming',
    'People': 'People & blogs',
    'Howto': 'Howto & Style',
    'Tech': 'Science & Technology',
    'Film': 'Film & Animation',
    'Movies': 'Film & Animation'    
}

In [10]:
df_raw['category'].replace(equivalents_category, inplace= True)

# Null Values

### Count of null values per column:

In [11]:
df_raw.isnull().sum()

rank                                  0
channel_name                          0
subscribers                           0
video_views                           0
category                              3
uploads                               0
country                             122
abbreviation                        122
channel_type                         30
video_views_rank                      1
country_rank                        116
channel_type_rank                    33
video_views_for_the_last_30_days     56
lowest_monthly_earnings               0
highest_monthly_earnings              0
lowest_yearly_earnings                0
highest_yearly_earnings               0
subscribers_for_last_30_days        337
created_year                          5
created_month                         5
created_date                          5
percentage_tertiary_education       123
population                          123
unemployment_rate                   123
urban_population                    123


### Replace nulls with 0:

- **'subscribers_for_last_30_days' ,**
- **'video_views_for_the_last_30_days'**

In [12]:
df_raw['subscribers_for_last_30_days'].fillna(0, inplace = True)
df_raw['subscribers_for_last_30_days'].count() 

995

In [13]:
df_raw['video_views_for_the_last_30_days'].fillna(0, inplace = True)
df_raw['video_views_for_the_last_30_days'].count()


995

### Detection that **the following columns match null values:**

- 'percentage_tertiary_education'
- 'population',
- 'unemployment_rate'
- 'urban_population'
- 'latitude'
- 'length'
- 'country'
- 'abbreviation'

There are 122 rows with null values that are repeated in 8 columns.

### Creation of a variable where I store the rows with those null values

In [14]:
null_values_df = df_raw[df_raw['abbreviation'].isnull()&df_raw['country'].isnull()&df_raw['percentage_tertiary_education'].isnull()&df_raw['population'].isnull()&df_raw['unemployment_rate'].isnull()&df_raw['urban_population'].isnull()&df_raw['latitude'].isnull()&df_raw['longitude'].isnull()]
null_values_df

rank                                       channel_name  subscribers  \
5       6                                              Music    119000000   
12     13                                             Gaming     93600000   
14     15                                          Goldmines     86900000   
38     39  LooLoo Kids - Nursery Rhymes and Children's Songs     54000000   
48     49                                            Badabun     46800000   
..    ...                                                ...          ...   
958   959                                     Troom Troom PT     12500000   
967   968                              Troom Troom Indonesia     12500000   
972   973                                   Hero Movies 2023     12400000   
985   986                                               TKOR     12400000   
986   987                                          ANNA KOVA     12400000   

      video_views          category  uploads country abbreviation  \
5    0.000000e+00             Music        0     NaN          NaN   
12   0.000000e+00            Gaming        0     NaN          NaN   
14   2.411823e+10  Film & Animation        1     NaN          NaN   
38   3.231243e+10             Music       11     NaN          NaN   
48   1.939805e+10     Entertainment        1     NaN          NaN   
..            ...               ...      ...     ...          ...   
958  4.384178e+09     Howto & Style     2738     NaN          NaN   
967  5.379684e+09    People & Blogs        8     NaN          NaN   
972  1.689091e+09    People & Blogs      689     NaN          NaN   
985  3.392023e+09         Education        0     NaN          NaN   
986  1.395959e+10    People & Blogs        1     NaN          NaN   

    channel_type  video_views_rank  ...  subscribers_for_last_30_days  \
5          Music         4057944.0  ...                           0.0   
12         Games         4057944.0  ...                           0.0   
14         Music         4056562.0  ...                           0.0   
38           NaN         3800129.0  ...                           0.0   
48         Music         4047729.0  ...                          75.0   
..           ...               ...  ...                           ...   
958        Howto            1667.0  ...                           0.0   
967       People         4057418.0  ...                           0.0   
972       People            6116.0  ...                           0.0   
985       People         4057944.0  ...                           0.0   
986         Film         4049634.0  ...                           0.0   

     created_year  created_month  created_date  percentage_tertiary_education  \
5          2013.0            Sep          24.0                            NaN   
12         2013.0            Dec          15.0                            NaN   
14         2006.0            Aug          15.0                            NaN   
38         2016.0            Nov          29.0                            NaN   
48         2007.0            Jul          21.0                            NaN   
..            ...            ...           ...                            ...   
958        2015.0            Apr          19.0                            NaN   
967        2020.0            Jul          29.0                            NaN   
972        2017.0            Feb          22.0                            NaN   
985        2006.0            Aug          16.0                            NaN   
986        2006.0            Jun          18.0                            NaN   

     population  unemployment_rate  urban_population  latitude longitude  
5           NaN                NaN               NaN       NaN       NaN  
12          NaN                NaN               NaN       NaN       NaN  
14          NaN                NaN               NaN       NaN       NaN  
38          NaN                NaN               NaN       NaN       NaN  
48          NaN           

### Creation of a new DataFrame without the rows containing the nulls

- This new DataFrame contains 873 entries

In [15]:
df =df_raw.drop(null_values_df.index)
df

rank                channel_name  subscribers   video_views  \
0       1                    T-Series    245000000  2.280000e+11   
1       2              YouTube Movies    170000000  0.000000e+00   
2       3                     MrBeast    166000000  2.836884e+10   
3       4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4       5                   SET India    159000000  1.480000e+11   
..    ...                         ...          ...           ...   
990   991               Natan por Aï¿     12300000  9.029610e+09   
991   992    Free Fire India Official     12300000  1.674410e+09   
992   993                       Panda     12300000  2.214684e+09   
993   994                 RobTopGames     12300000  3.741235e+08   
994   995                Make Joke Of     12300000  2.129774e+09   

             category  uploads         country abbreviation   channel_type  \
0               Music    20082           India           IN          Music   
1    Film & Animation        1   United States           US          Games   
2       Entertainment      741   United States           US  Entertainment   
3           Education      966   United States           US      Education   
4               Shows   116536           India           IN  Entertainment   
..                ...      ...             ...          ...            ...   
990            Sports     1200          Brazil           BR  Entertainment   
991    People & Blogs     1500           India           IN          Games   
992            Gaming     2452  United Kingdom           GB          Games   
993            Gaming       39          Sweden           SE          Games   
994            Comedy       62           India           IN         Comedy   

     video_views_rank  ...  subscribers_for_last_30_days  created_year  \
0                 1.0  ...                     2000000.0        2006.0   
1           4055159.0  ...                           0.0        2006.0   
2                48.0  ...                     8000000.0        2012.0   
3                 2.0  ...                     1000000.0        2006.0   
4                 3.0  ...                     1000000.0        2006.0   
..                ...  ...                           ...           ...   
990             525.0  ...                      700000.0        2017.0   
991            6141.0  ...                      300000.0        2018.0   
992          129005.0  ...                        1000.0        2006.0   
993           35112.0  ...                      100000.0        2012.0   
994            4568.0  ...                      100000.0        2017.0   

     created_month  created_date  percentage_tertiary_education    population  \
0              Mar          13.0                           28.1  1.366418e+09   
1              Mar           5.0                           88.2  3.282395e+08   
2              Feb          20.0                           88.2  3.282395e+08   
3              Sep           1.0                           88.2  3.282395e+08   
4              Sep          20.0                           28.1  1.366418e+09   
..             ...           ...                            ...           ...   
990            Feb          12.0                           51.3  2.125594e+08   
991            Sep          14.0                           28.1  1.366418e+09   
992            Sep          11.0                           60.0  6.683440e+07   
993            May           9.0                           67.0  1.028545e+07   
994            Aug           1.0                           28.1  1.366418e+09   

     unemployment_rate  urban_population   latitude  longitude  
0                 5.36       471031528.0  20.593684  78.962880  
1                14.70       270663028.0  37.090240 -95.712891  
2                14.70       270663028.0  37.090240 -95.712891  
3                14.70       270663028.0  37.090240 -95.712891  
4                 5.36       471031528.0  20.593684  78.962880  

In [16]:
df.isnull().sum()

rank                                0
channel_name                        0
subscribers                         0
video_views                         0
category                            0
uploads                             0
country                             0
abbreviation                        0
channel_type                        7
video_views_rank                    1
country_rank                        1
channel_type_rank                   9
video_views_for_the_last_30_days    0
lowest_monthly_earnings             0
highest_monthly_earnings            0
lowest_yearly_earnings              0
highest_yearly_earnings             0
subscribers_for_last_30_days        0
created_year                        3
created_month                       3
created_date                        3
percentage_tertiary_education       1
population                          1
unemployment_rate                   1
urban_population                    1
latitude                            1
longitude   

### **Removal of the 'channel_type' Column**

- This column is equivalent to the 'category' column.

In [17]:
df.drop(columns='channel_type', inplace = True)

### Exploration of the null values in the following columns:

- percentage_tertiary_education
- population
- unemployment_rate
- urban_population
- latitude
- longitude

In [18]:
df[['percentage_tertiary_education','population', 'unemployment_rate','urban_population','latitude','longitude']]

percentage_tertiary_education    population  unemployment_rate  \
0                             28.1  1.366418e+09               5.36   
1                             88.2  3.282395e+08              14.70   
2                             88.2  3.282395e+08              14.70   
3                             88.2  3.282395e+08              14.70   
4                             28.1  1.366418e+09               5.36   
..                             ...           ...                ...   
990                           51.3  2.125594e+08              12.08   
991                           28.1  1.366418e+09               5.36   
992                           60.0  6.683440e+07               3.85   
993                           67.0  1.028545e+07               6.48   
994                           28.1  1.366418e+09               5.36   

     urban_population   latitude  longitude  
0         471031528.0  20.593684  78.962880  
1         270663028.0  37.090240 -95.712891  
2         270663028.0  37.090240 -95.712891  
3         270663028.0  37.090240 -95.712891  
4         471031528.0  20.593684  78.962880  
..                ...        ...        ...  
990       183241641.0 -14.235004 -51.925280  
991       471031528.0  20.593684  78.962880  
992        55908316.0  55.378051  -3.435973  
993         9021165.0  60.128161  18.643501  
994       471031528.0  20.593684  78.962880  

[873 rows x 6 columns]

### **Replacement of null values with 0 in the following columns:**

- video_views_rank
- country_rank
- channel_type_rank
- percentage_tertiary_education
- population
- unemployment_rate
- urban_population
- latitude
- longitude

In [19]:
columns_to_convert = df[['video_views_rank','country_rank','channel_type_rank','percentage_tertiary_education', 'population', 'unemployment_rate','urban_population','latitude','longitude']]

for col in columns_to_convert:
    df[col] = df[col].fillna(0)

print(df.isnull().sum())

rank                                0
channel_name                        0
subscribers                         0
video_views                         0
category                            0
uploads                             0
country                             0
abbreviation                        0
video_views_rank                    0
country_rank                        0
channel_type_rank                   0
video_views_for_the_last_30_days    0
lowest_monthly_earnings             0
highest_monthly_earnings            0
lowest_yearly_earnings              0
highest_yearly_earnings             0
subscribers_for_last_30_days        0
created_year                        3
created_month                       3
created_date                        3
percentage_tertiary_education       0
population                          0
unemployment_rate                   0
urban_population                    0
latitude                            0
longitude                           0
dtype: int64

### Data type

Making adjustments to column data types to optimize resource utilization and facilitate analysis:

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 873 entries, 0 to 994
Data columns (total 26 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   rank                              873 non-null    int64  
 1   channel_name                      873 non-null    object 
 2   subscribers                       873 non-null    int64  
 3   video_views                       873 non-null    float64
 4   category                          873 non-null    object 
 5   uploads                           873 non-null    int64  
 6   country                           873 non-null    object 
 7   abbreviation                      873 non-null    object 
 8   video_views_rank                  873 non-null    float64
 9   country_rank                      873 non-null    float64
 10  channel_type_rank                 873 non-null    float64
 11  video_views_for_the_last_30_days  873 non-null    float64
 12  lowest_m

### Changing the data type from 'object' to 'category'

- 'category' 
- 'country'
- 'abbreviation'

In [21]:
columns_to_category = df[['category','country', 'abbreviation']]

for col in columns_to_category:
    df[col] = df[col].astype('category')
print(df.dtypes[['category','country', 'abbreviation']])

category        category
country         category
abbreviation    category
dtype: object


### Changing the data type from 'float' to 'int64' for the columns

These columns are in float format, representing information is numeric and integer data. Therefore, I am converting them to int64 format for better clarity and intuitive handling.
- 'video_views'
- 'video_views_rank '
- 'country_rank'
- 'channel_type_rank'
- 'video_views_for_the_last_30_days'
- 'subscribers_for_last_30_days'
- 'created_year'
- 'created_date'
- 'population'
- 'urban_population'

In [22]:
columns_to_Int64 = df[['video_views','video_views_rank','country_rank','channel_type_rank','video_views_for_the_last_30_days','subscribers_for_last_30_days','created_year','created_date','population','urban_population']]

for col in columns_to_Int64:
    df[col] = df[col].astype('Int64')

## New column for full date

### Creation **a dictionary of equivalences for months**

- The 'created_month' column contains the abbreviations of the months of the year, it is necessary to convert it to its numerical representation

In [23]:
month = {
   'Jan': 1, 'Feb': 2, 'Mar': 3, 'Apr': 4, 'May': 5, 'Jun': 6,
    'Jul': 7, 'Aug': 8, 'Sep': 9, 'Oct': 10, 'Nov': 11, 'Dec': 12
}

### Replacing 'created_month' values with dictionary values

In [24]:
df['created_month'] = df['created_month'].map(month)

### Change type to Int64

In [25]:
df['created_month'] = df['created_month'].astype('Int64')

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 873 entries, 0 to 994
Data columns (total 26 columns):
 #   Column                            Non-Null Count  Dtype   
---  ------                            --------------  -----   
 0   rank                              873 non-null    int64   
 1   channel_name                      873 non-null    object  
 2   subscribers                       873 non-null    int64   
 3   video_views                       873 non-null    Int64   
 4   category                          873 non-null    category
 5   uploads                           873 non-null    int64   
 6   country                           873 non-null    category
 7   abbreviation                      873 non-null    category
 8   video_views_rank                  873 non-null    Int64   
 9   country_rank                      873 non-null    Int64   
 10  channel_type_rank                 873 non-null    Int64   
 11  video_views_for_the_last_30_days  873 non-null    Int64   

### Drop null values in columns 'created_year', 'created_date', and 'created_month' 

In [27]:
null_values_date = df[['created_date','created_month', 'created_year']]

for col in null_values_date:
    df= df.dropna(subset = [col], axis = 0)

### Creation of 'date' column

- Creation of a variable where the columns 'created_year', 'created_month', and 'created_date' are renamed to a format that can process pd.to_datetime
- Creaction of the 'date' column applying the datetime type

In [28]:
columns_date = df[['created_date','created_month', 'created_year']].rename(columns= {
    'created_date': 'day',
    'created_month': 'month',
    'created_year':'year'
})
df['date']=pd.to_datetime(columns_date, errors = 'coerce')

In [29]:
# Checking
df[['date', 'created_date','created_month', 'created_year']]

date  created_date  created_month  created_year
0   2006-03-13            13              3          2006
1   2006-03-05             5              3          2006
2   2012-02-20            20              2          2012
3   2006-09-01             1              9          2006
4   2006-09-20            20              9          2006
..         ...           ...            ...           ...
990 2017-02-12            12              2          2017
991 2018-09-14            14              9          2018
992 2006-09-11            11              9          2006
993 2012-05-09             9              5          2012
994 2017-08-01             1              8          2017

[870 rows x 4 columns]

### Drop columns

- 'created_year'
- 'created_month'
- 'created_date'

In [30]:
df.drop(columns= ['created_year','created_month','created_date'], inplace = True)

# Outliers

In [31]:
df['date'].describe()

C:\Users\irism\AppData\Local\Temp\ipykernel_7208\801468514.py:1: FutureWarning: Treating datetime data as categorical rather than numeric in `.describe` is deprecated and will be removed in a future version of pandas. Specify `datetime_is_numeric=True` to silence this warning and adopt the future behavior now.
  df['date'].describe()


count                     870
unique                    782
top       2009-05-12 00:00:00
freq                        6
first     1970-01-01 00:00:00
last      2022-06-27 00:00:00
Name: date, dtype: object

In [32]:
df.loc[df['date']== '1970-01-01']

rank channel_name  subscribers  video_views         category  uploads  \
101   102      YouTube     36300000   3010784935  News & Politics      744   

           country abbreviation  video_views_rank  country_rank  ...  \
101  United States           US              2860            32  ...   

     lowest_yearly_earnings  highest_yearly_earnings  \
101                 63300.0                1000000.0   

     subscribers_for_last_30_days  percentage_tertiary_education  population  \
101                        300000                           88.2   328239523   

     unemployment_rate  urban_population  latitude  longitude       date  
101               14.7         270663028  37.09024 -95.712891 1970-01-01  

[1 rows x 24 columns]

### An outlier is found in the date, so I remove that row

In [33]:
df=df.drop(df.loc[df['date']== '1970-01-01'].index)

# Exploratory Data Analysis


### Exploring 'video_views_for_the_last_30_days' and 'subscribers_for_last_30_days' columns

In [34]:
df[['date','video_views_for_the_last_30_days','subscribers_for_last_30_days']]

date  video_views_for_the_last_30_days  subscribers_for_last_30_days
0   2006-03-13                        2258000000                       2000000
1   2006-03-05                                12                             0
2   2012-02-20                        1348000000                       8000000
3   2006-09-01                        1975000000                       1000000
4   2006-09-20                        1824000000                       1000000
..         ...                               ...                           ...
990 2017-02-12                         552513000                        700000
991 2018-09-14                          64735000                        300000
992 2006-09-11                             67035                          1000
993 2012-05-09                           3871000                        100000
994 2017-08-01                          24000000                        100000

[869 rows x 3 columns]

### Eliminate these columns because they do not have very relevant information, the data does not belong to the same time period

In [35]:
df.drop(columns=['video_views_for_the_last_30_days','subscribers_for_last_30_days'], inplace = True)

### Elimination of column [1] that belongs to YouTube movies, which is something from the platform itself, not an external channel.

In [36]:
df.drop[1]

TypeError: 'method' object is not subscriptable

### Descriptive Analysis

- 'subscribers'
- 'video_views'

In [ ]:
df[['rank', 'channel_name', 'subscribers', 'video_views', 'uploads']].head(10)

In [ ]:
.sort_values(by='subscribers', ascending=False)

In [ ]:
df[['subscribers','video_views','uploads']].describe().round(1)

In [ ]:
sns.barplot(data=df, x='category', y='subscribers', color='blue', label='subscribers');

- 'lowest_monthly_earnings'
- 'highest_monthly_earnings'
- 'lowest_yearly_earnings'
- 'percentage_tertiary_education'
- 'population'
- 'unemployment_rate'
- 'urban_population'

# Youtube Analytics



### What is the distribution of subscribers among the analyzed YouTube channels?

In [ ]:
df['subscribers'].describe().round(1)

In [ ]:
sns.histplot(data=df, x='subscribers', bins=25, color='blue', kde=True);

### How does the total video views ('video_views') correlate with the number of subscribers ('subscribers')?

In [ ]:
df[['subscribers', 'video_views']].corr()

# No aporta valor las suscripciones ni visitas en el ultimo mes
# cambiar la pregunta
### What is the growth rate of subscribers in the last month for the channels?

In [ ]:
sns.lineplot(data=df, x='date', y='video_views');

### What is the most common content category among successful channels?

In [ ]:
df['category'].value_counts()

In [ ]:
sns.countplot(data=df, y='category', order=df['category'].value_counts().index, palette='viridis');

### What are the countries with the highest number of subscribers on YouTube?

In [ ]:
top_10_subscribers = df.groupby('country')['subscribers'].sum().reset_index()
top_10_subscribers = top_10_subscribers.sort_values(by='subscribers', ascending=False)
top_10_subscribers=top_10_subscribers.head(10)
top_10_subscribers['subscribers'] = top_10_subscribers['subscribers']/1000

sns.barplot(data= top_10_subscribers , x='subscribers', y='country', estimator=sum, palette='Set3', order=top_10_subscribers['country']);
plt.xlabel('Subscribers (1 billion)');  # Cambiar la etiqueta del eje x

In [ ]:
sns.lmplot(data=df, x='subscribers', y='video_views', scatter_kws={'alpha':0.3}, line_kws={'color': 'red'})
